core/src/main/java/org/elasticsearch/action/index/IndexResponse.java

```diff
@@ -107,15 +88,10 @@ public static void parseXContentFields(XContentParser parser, Builder context) t
      */
     public static class Builder extends DocWriteResponse.Builder {
 
-        private boolean created = false;
-
-        public void setCreated(boolean created) {
-            this.created = created;
-        }
-
         @Override
         public IndexResponse build() {
-            IndexResponse indexResponse = new IndexResponse(shardId, type, id, seqNo, primaryTerm, version, created);
+            IndexResponse indexResponse = new IndexResponse(shardId, type, id, seqNo, primaryTerm, version,
+                    result == Result.CREATED ? true : false);
```

Comment:
Same deal here as on Delete. I think you should pass the `result` into the ctor and `assert` that it is sane.